content/guides/docker-concepts/running-containers/sharing-local-files.md

```diff
@@ -0,0 +1,211 @@
+
+Imagine you're a developer working on a project. You have a source directory on your development machine where your code resides. When you compile or build your code, the generated artifacts (compiled code, executables, images, etc.) are saved in a separate subdirectory within your source directory. Let's call this subdirectory  `/HOST/PATH`. Now you want these build artifacts to be accessible within a Docker container running your application. Additionally, you want the container to automatically access the latest build artifacts whenever you rebuild your code.
+
+Here’s a way how you use `docker run` to start a container using a bind mount and map it to the container file location.
+
+```console
+$ docker run -v /HOST/PATH:/CONTAINER/PATH -it nginx
+```
+
+The `--mount` offers more advanced features and granular control, making it suitable for complex mount scenarios or production deployments. If you use `--mount` to bind-mount a file or directory that doesn't yet exist on the Docker host, the `docker run` command doesn't automatically create it for you but generates an error.
+
+```console
+$ docker run --mount type=bind,source=/HOST/PATH,target=/CONTAINER/PATH,readonly nginx
+```
+
+> It is recommended to use the `--mount` syntax instead of `-v`. It provides better control over the mounting process and avoids potential issues with missing directories.
+
+### File permissions for Docker access to host files
+
+When using bind mounts, it's crucial to ensure that Docker has the necessary permissions to access the host directory. To grant read/write access, you can use the `:ro` flag (read-only) or `:rw` (read-write) with the `-v` or `--mount` flag during container creation.
+For example, the following command grants read-write access permission.
+
+```console
+$ docker run -v HOST-DIRECTORY:/CONTAINER-DIRECTORY:rw nginx
+```
+
+Now you might ask, when are read-only bind mounts useful? Read-only bind mounts are particularly suitable for the development phase. They allow the container to access configuration files or data without accidentally modifying them on the host machine. This is beneficial for maintaining a clean development environment. Additionally, read-only bind mounts can be used to prevent unauthorized changes within the container, especially when handling sensitive data. This improves security by ensuring the integrity of the data within the container.
+
+> **Synchronised File Share**
+>
+> As your codebase grows larger, traditional methods of file sharing like bind mounts may become inefficient or slow, especially in development environments where frequent access to files is necessary. Synchronized file shares (available in Docker Desktop for Mac version 4.27 and later) improve bind mount performance by leveraging synchronized filesystem caches. This optimization ensures that file access between the host and virtual machine (VM) is fast and efficient.
```

Comment:
Recommend not listing the version here and instead linking to the more detailed topic.

```suggestion
> As your codebase grows larger, traditional methods of file sharing like bind mounts may become inefficient or slow, especially in development environments where frequent access to files is necessary. [Synchronized file shares](/desktop/synchronized-file-sharing/) improve bind mount performance by leveraging synchronized filesystem caches. This optimization ensures that file access between the host and virtual machine (VM) is fast and efficient.
```